generator/restriction_collector.cpp

```diff
@@ -65,34 +70,31 @@ bool RestrictionCollector::IsValid() const
                  [](Restriction const & r) { return !r.IsValid(); }) == end(m_restrictions);
 }
 
-bool RestrictionCollector::ParseFeatureId2OsmIdsMapping(string const & featureIdToOsmIdsPath)
+bool RestrictionCollector::ParseOsmIdToFeatureIdMapping(string const & osmIdsToFeatureIdPath)
 {
-  ifstream featureIdToOsmIdsStream(featureIdToOsmIdsPath);
-  if (featureIdToOsmIdsStream.fail())
+  gen::OsmID2FeatureID osmIdsToFeatureIds;
+  try
+  {
+    FileReader reader(osmIdsToFeatureIdPath);
+    ReaderSource<FileReader> src(reader);
+    osmIdsToFeatureIds.Read(src);
+  }
+  catch (FileReader::Exception const & e)
+  {
+    LOG(LWARNING, ("Exception while reading file:", osmIdsToFeatureIdPath, ". Msg:", e.Msg()));
     return false;
+  }
 
-  string line;
-  while (getline(featureIdToOsmIdsStream, line))
+  vector<gen::OsmIdAndFeatureId> const & mapping = osmIdsToFeatureIds.GetData();
```

Comment:
I suggest you to implement ForEach() in OsmId2FeatureId class, instead of this raw and hacky GetData().